dandischema/metadata.py

```diff
@@ -245,6 +288,8 @@ def migrate(
         if "schemaKey" not in obj:
             obj["schemaKey"] = "Dandiset"
         obj["schemaVersion"] = to_version
+    if version2tuple(schema_version) < version2tuple("0.7.0"):
+        pass  # TODO: do necessary changes so that old metadata (before 0.7.0) looks like new (0.7.0)
```

Comment:
Remove these two lines as we should not have made any changes in the data model.

```suggestion
```

Reply:
i would agree. that's a question for @yarikoptic - he felt this was more of a minor change as the json schema did have several changes just stylistically.

Reply:
I am actually assessing the impact of the changes of JSON schema on the serialization of the models now. They impact can actually be small.

Reply:
@satra @jwodder @yarikoptic I can confirm that there is no changes in the data models. Serialization of object instances of the models should be the same with Pydantic V2 as with Pydantic V1. The changes are in the schema of those models.

@satra The lines you requested to be removed are now removed.

Can you all now confirm that the new version of the schema should be 0.6.5 instead of 0.7.0 since the changes in the schema are not breaking changes even though there are quite a few of them and widespread?

Reply:
yes, to me schema version can be 0.6.5 and the library will be a minor release with all the changes.

Reply:
> yes, to me schema version can be 0.6.5 and the library will be a minor release with all the changes.

schema version is now `0.6.5` instead of `0.7.0`.

Reply:
What else is needed to be done before merging? May be merging on a Friday is not a good idea?

Reply:
i'm going to move this back to ready for review. and i'll hit release first thing on monday morning.